In [46]:
import pandas as pd
import sqlite3

# The Movie Database (tmdb.movies.csv)

In [14]:
df_movies = pd.read_csv('./data/tmdb.movies.csv', index_col=0)

In [92]:
df_movies.sort_values(by='release_date', ascending = False).head(10)

genre_ids      id original_language  \
26057  [27, 80, 80, 80, 80, 80, 80]  570704                en   
24265                   [10749, 18]  428836                en   
24892                          [99]  541577                en   
24819                          [18]  481880                en   
24297                          [18]  415085                en   
24003                [18, 9648, 53]  411144                en   
25006                          [99]  500850                es   
24612                          [99]  541576                en   
24691                  [18, 28, 80]  547590                en   
25012                          [99]  523994                en   

                           original_title  popularity release_date  \
26057                   Murdery Christmas       0.840   2020-12-25   
24265                             Ophelia       8.715   2019-06-28   
24892             This Changes Everything       3.955   2019-06-28   
24819                       Trial by Fire       4.480   2019-05-17   
24297            All Creatures Here Below       8.316   2019-05-17   
24003  We Have Always Lived in the Castle      14.028   2019-05-17   
25006                El silencio de otros       3.299   2019-05-08   
24612                   Meeting Gorbachev       5.870   2019-05-03   
24691                          El Chicano       5.274   2019-05-03   
25012                            Hesburgh       3.262   2019-04-26   

                                    title  vote_average  vote_count  
26057                   Murdery Christmas           0.0           1  
24265                             Ophelia           0.0           4  
24892             This Changes Everything           0.0           1  
24819                       Trial by Fire           7.0           3  
24297            All Creatures Here Below           5.0           5  
24003  We Have Always Lived in the Castle           5.2          24  
25006               The Silence of Others           8.5          15  
24612                   Meeting Gorbachev          10.0           3  
24691                          El Chicano           9.0           1  
25012                            Hesburgh          10.0           1

In [15]:
df_movies.head()

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [57]:
df_movies.shape


(26517, 9)

In [58]:
df_movies.columns

Index(['genre_ids', 'id', 'original_language', 'original_title', 'popularity',
       'release_date', 'title', 'vote_average', 'vote_count'],
      dtype='object')

In [59]:
df_movies.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


In [64]:
df_movies['genre_ids'].value_counts()

[99]                         3700
[]                           2479
[18]                         2268
[35]                         1660
[27]                         1145
                             ... 
[36, 18, 53, 9648]              1
[16, 28, 35, 14, 12]            1
[10770, 18, 80]                 1
[28, 53, 18, 9648]              1
[12, 14, 27, 28, 53, 878]       1
Name: genre_ids, Length: 2477, dtype: int64

### Genre Keys
We can match these with the numbers in "genre_ids" column
(per https://www.themoviedb.org/talk/5daf6eb0ae36680011d7e6ee)
- Action          28
- Adventure       12
- Animation       16
- Comedy          35
- Crime           80
- Documentary     99
- Drama           18
- Family          10751
- Fantasy         14
- History         36
- Horror          27
- Music           10402
- Mystery         9648
- Romance         10749
- Science Fiction 878
- TV Movie        10770
- Thriller        53
- War             10752
- Western         37

# Rotten Tomatoes (rt.movie_info.tsv)

In [86]:
df_info = pd.read_csv('./data/rt.movie_info.tsv', delimiter='\t')

In [87]:
df_info.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [90]:
df_info['theater_date'] = pd.to_datetime(df_info['theater_date'])

df_info.sort_values(by='theater_date', ascending = False).head(10)

id                                           synopsis rating  \
973   1264  The Super centers on the mysterious disappeara...      R   
813   1050  In the tradition of Amblin classics where fant...     PG   
1023  1325  From the outer reaches of space to the small-t...      R   
102    131  No Sesame. All Street. THE HAPPYTIME MURDERS i...      R   
1430  1845  A group of wealthy boys in Los Angeles during ...      R   
1064  1374                                 Cameron Post (Chlo     NR   
1201  1541  In the heatwarming live action adventure "Disn...     PG   
319    419  When teens mysteriously develop powerful new a...  PG-13   
446    585  Denzel Washington returns to one of his signat...      R   
1003  1297  Based on true events, WOMAN WALKS AHEAD tells ...      R   

                                                  genre             director  \
973                                Mystery and Suspense         Stephan Rick   
813   Comedy|Kids and Family|Science Fiction and Fan...             Eli Roth   
1023  Action and Adventure|Horror|Science Fiction an...          Shane Black   
102                         Action and Adventure|Comedy         Brian Henson   
1430                         Drama|Mystery and Suspense            James Cox   
1064                                              Drama      Desiree Akhavan   
1201        Action and Adventure|Comedy|Kids and Family         Marc Forster   
319    Mystery and Suspense|Science Fiction and Fantasy  Jennifer Yuh Nelson   
446           Action and Adventure|Mystery and Suspense        Antoine Fuqua   
1003                                              Drama        Susanna White   

                                 writer theater_date      dvd_date currency  \
973                                 NaN   2018-10-19  Dec 18, 2018      NaN   
813                         Eric Kripke   2018-09-21  Nov 27, 2018      NaN   
1023            Shane Black|Fred Dekker   2018-09-14  Nov 27, 2018      NaN   
102    Todd Berger|Dee Austin Robertson   2018-08-24   Dec 4, 2018      NaN   
1430          Captain Mauzner|James Cox   2018-08-17  Sep 18, 2018      NaN   
1064  Desiree Akhavan|Cecilia Frugiuele   2018-08-03   Dec 3, 2018      NaN   
1201                    Alex Ross Perry   2018-08-03   Nov 6, 2018      NaN   
319                          Chad Hodge   2018-08-03  Oct 30, 2018      NaN   
446                        Richard Wenk   2018-07-20  Dec 11, 2018      NaN   
1003                      Steven Knight   2018-06-29  Aug 28, 2018      NaN   

     box_office      runtime                studio  
973         NaN   90 minutes                   NaN  
813         NaN  104 minutes    Universal Pictures  
1023        NaN          NaN                   NaN  
102         NaN          NaN              STXfilms  
1430        NaN  100 minutes                   NaN  
1064        NaN   90 minutes              FilmRise  
1201        NaN          NaN  Walt Disney Pictures  
319         NaN  105 minutes      20th Century Fox  
446         NaN  129 minutes     Columbia Pictures  
1003        NaN  102 minutes                   NaN

In [30]:
df_info.shape

(1560, 12)

In [31]:
df_info.columns

Index(['id', 'synopsis', 'rating', 'genre', 'director', 'writer',
       'theater_date', 'dvd_date', 'currency', 'box_office', 'runtime',
       'studio'],
      dtype='object')

In [32]:
df_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


# The Numbers (tn.movie_budgets.csv)

In [34]:
df_budget = pd.read_csv('./data/tn.movie_budgets.csv', index_col=0)

In [93]:
df_budget['release_date'] = pd.to_datetime(df_budget['release_date'])

In [97]:
df_budget.sort_values(by='release_date', ascending=False).head(10)

release_date                   movie production_budget domestic_gross  \
id                                                                         
95   2020-12-31                Moonfall      $150,000,000             $0   
6    2020-12-31  Hannibal the Conqueror       $50,000,000             $0   
36   2020-02-21        Call of the Wild       $82,000,000             $0   
81   2019-12-31        Army of the Dead       $90,000,000             $0   
16   2019-12-31                     Eli       $11,000,000             $0   
72   2019-12-31                     355       $75,000,000             $0   
44   2019-12-31        Down Under Cover       $40,000,000             $0   
30   2019-12-31                  Reagan       $25,000,000             $0   
13   2019-12-31              Rogue City       $13,000,000             $0   
8    2019-11-22      The Rhythm Section       $50,000,000             $0   

   worldwide_gross  
id                  
95              $0  
6               $0  
36              $0  
81              $0  
16              $0  
72              $0  
44              $0  
30              $0  
13              $0  
8               $0

In [35]:
df_budget.head()

release_date                                        movie  \
id                                                              
1   Dec 18, 2009                                       Avatar   
2   May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
3    Jun 7, 2019                                 Dark Phoenix   
4    May 1, 2015                      Avengers: Age of Ultron   
5   Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

   production_budget domestic_gross worldwide_gross  
id                                                   
1       $425,000,000   $760,507,625  $2,776,345,279  
2       $410,600,000   $241,063,875  $1,045,663,875  
3       $350,000,000    $42,762,350    $149,762,350  
4       $330,600,000   $459,005,868  $1,403,013,963  
5       $317,000,000   $620,181,382  $1,316,721,747

In [36]:
df_budget.shape

(5782, 5)

In [37]:
df_budget.columns

Index(['release_date', 'movie', 'production_budget', 'domestic_gross',
       'worldwide_gross'],
      dtype='object')

In [38]:
df_budget.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5782 entries, 1 to 82
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   release_date       5782 non-null   object
 1   movie              5782 non-null   object
 2   production_budget  5782 non-null   object
 3   domestic_gross     5782 non-null   object
 4   worldwide_gross    5782 non-null   object
dtypes: object(5)
memory usage: 271.0+ KB


# Box Office Mojo (bom.movie_gross.csv)



In [40]:
df_gross = pd.read_csv('./data/bom.movie_gross.csv')

In [41]:
df_gross.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [98]:
df_gross.sort_values(by='year', ascending=False)

title studio  domestic_gross foreign_gross  \
3386               An Actor Prepares  Grav.          1700.0           NaN   
3183             On the Basis of Sex  Focus      24600000.0      13600000   
3176          Tyler Perry's Acrimony    LGF      43500000.0       2900000   
3177             Mary Queen of Scots  Focus      16500000.0      29900000   
3178  The Possession of Hannah Grace   SGem      14800000.0      28200000   
...                              ...    ...             ...           ...   
220                       After.Life  Anch.        109000.0       1900000   
221                       Cairo Time    IFC       1600000.0        391000   
222                          Flipped     WB       1800000.0           NaN   
223                        Guzaarish    UTV       1000000.0        695000   
0                        Toy Story 3     BV     415000000.0     652000000   

      year  
3386  2018  
3183  2018  
3176  2018  
3177  2018  
3178  2018  
...    ...  
220   2010  
221   2010  
222   2010  
223   2010  
0     2010  

[3387 rows x 5 columns]

In [42]:
df_gross.shape

(3387, 5)

In [43]:
df_gross.columns

Index(['title', 'studio', 'domestic_gross', 'foreign_gross', 'year'], dtype='object')

In [45]:
df_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


# IM.DB

In [67]:
conn = sqlite3.Connection("./data/im.db")

In [74]:
q0 = """
SELECT *
FROM movie_ratings
;
"""

df_ratings = pd.read_sql(q0, conn)

In [75]:
df_ratings.shape

(73856, 3)

In [76]:
df_ratings.head()

movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [71]:
q2 = """
SELECT *
FROM movie_basics
;
"""

df_movie_basics = pd.read_sql(q2, conn)


In [72]:
df_movie_basics.shape

(146144, 6)

In [73]:
df_movie_basics.head()

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [105]:
pd.read_sql("""
SELECT *
FROM movie_basics
WHERE
    start_year > 2010 AND
    start_year <= 2022
ORDER BY
    start_year DESC
;
"""

,conn)


movie_id                               primary_title  \
0       tt10042446                 Untitled Disney Marvel Film   
1       tt10064536                 Untitled Disney Marvel Film   
2       tt10108854             The Hunchback of the Lighthouse   
3       tt10251718                       Corazones en Llamas 5   
4       tt10288908                          The Weary Traveler   
...            ...                                         ...   
134279   tt9844420                                   Star High   
134280   tt9856614                   Kanun il sangue e l'onore   
134281   tt9878416                     Xing Fu De Xiang Ri Kui   
134282   tt9896506                   Six Hours With Terrorists   
134283   tt9898682  Vengeance Douloureuse 1: la colère est née   

                                    original_title  start_year  \
0                      Untitled Disney Marvel Film        2022   
1                      Untitled Disney Marvel Film        2022   
2                  The Hunchback of the Lighthouse        2022   
3                            Corazones en Llamas 5        2022   
4                               The Weary Traveler        2022   
...                                            ...         ...   
134279                                   Star High        2011   
134280                   Kanun il sangue e l'onore        2011   
134281                     Xing Fu De Xiang Ri Kui        2011   
134282                   Six Hours With Terrorists        2011   
134283  Vengeance Douloureuse 1: la colère est née        2011   

        runtime_minutes       genres  
0                   NaN       Action  
1                   NaN       Action  
2                 120.0        Drama  
3                   NaN       Action  
4                   NaN       Horror  
...                 ...          ...  
134279             66.0       Sci-Fi  
134280              NaN  Documentary  
134281              NaN        Drama  
134282              NaN        Drama  
134283             76.0       Horror  

[134284 rows x 6 columns]

In [56]:
conn.close()

# Problem & Questions

Business Problem:

You are charged with exploring what types of films are currently doing the best at the box office. You must then translate those findings into actionable insights that the head of Microsoft's new movie studio can use to help decide what type of films to create.

Questions:

- What genre of movies are doing well currently? 
    - are movie franchises, i.e. star wars, marvel, harry potter, etc. any better than one-offs when measuring success
- what does "doing well" mean?
    - presumably, total gross > production budget. The Numbers (data source) includes domestic & international in Worldwide Gross.   
    - based on the data we have, we could measure by worldwide gross, imdb rating, The Movie Database popularity
    - both are important for a new studio, probably lean toward measuring success monetarily. need some easy wins 
        - maybe measure against MSFT financial position?
- what actors are in those successful movies?
- what writers/directors are responsible for the movies?
- is rating significant?
- does movie length have any impact?




Data Clearning Thoughts:

- We'll need to decide what "doing well currently" means. We'll be able to eliminate a lot of bloat by ignoring things that were released too long ago to be relevant. Maybe we start at movies released in the last 10-15 years?
        - Doesn't appear these sources line up perfectly for the date they were created. Maybe better to set our acceptable date range than "last X years".

- probably want to identify the categories we want from each csv/db and then create one dataframe to work with. In addition to making comparisons easier, I think that's our most realistic path to solving the file size issue git has with im.db.

- the im.db file has the most records by far of any of our data sources. I think maybe we could make our starting position the movie_basics table (146k rows), and then add columns from the other sources to it and drop records for which we don't have enough info.

- remember we don't have to use all data sources--don't want to create problems with unknown values if a data file is super short (rotten tomatoes). Will see what happens with that.


Taking a stab at the columns in our ideal dataframe:
   - Movie Title
   - Genre(s)
   - Release Year
   - Rating (R/PG-13/etc)
   - Total Gross
   - Production Budget
   - Associated Actors/Star/TBD
   - Writer
   - Director
   - Run Time
   - IMDB Rating
   - The Movie Database Popularity (still need to understand what this is)
   - The Movie Database Vote Count